solidity/contracts/igps/GasOverheadIgp.sol

```diff
@@ -0,0 +1,179 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+// ============ Internal Imports ============
+import {IInterchainGasPaymaster} from "../../interfaces/IInterchainGasPaymaster.sol";
+// ============ External Imports ============
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+/**
+ * @notice An IGP that adds configured gas overheads to gas amounts and forwards
+ * calls to an "inner" IGP.
+ * @dev The intended use of this contract is to store overhead gas amounts for destination
+ * domains, e.g. Mailbox and/or ISM gas usage, such that users of this IGP are only required
+ * to specify the gas amount used by their own applications.
+ */
+contract GasOverheadIgp is IInterchainGasPaymaster, OwnableUpgradeable {
+    // ============ Public Storage ============
+
+    /// @notice The IGP that is called when paying for or quoting gas
+    /// after applying overhead gas amounts.
+    IInterchainGasPaymaster public innerIgp;
+
+    /// @notice Destination domain => overhead gas amount on that domain.
+    mapping(uint32 => uint256) public destinationGasOverhead;
+
+    // ============ Upgrade Gap ============
+
+    // gap for upgrade safety
+    uint256[48] private __GAP;
+
+    // ============ Events ============
+
+    /**
+     * @notice Emitted when the innerIgp is set.
+     * @param innerIgp The new innerIgp.
+     */
+    event InnerIgpSet(address innerIgp);
+
+    /**
+     * @notice Emitted when an entry in the destinationGasOverhead mapping is set.
+     * @param domain The destination domain.
+     * @param gasOverhead The gas overhead amount on that domain.
+     */
+    event DestinationGasOverheadSet(uint32 indexed domain, uint256 gasOverhead);
+
+    // ============ Constructor ============
+
+    constructor(address _innerIgp) {
+        initialize(_innerIgp); // allows contract to be used without proxying
+    }
+
+    // ============ Initializers ============
+
+    /**
+     * @notice Initializes the contract.
+     * @param _innerIgp The innerIgp.
+     */
+    function initialize(address _innerIgp) public initializer {
+        __Ownable_init();
```

Comment:
we've had problems with proxying ownership here
could you pls add an explicit `address owner` arg here and transfer ownership
happy for the constructor to pass `msg.sender` for non proxied setups